keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1892 @@
+
+### Topology Manager
+
+The Topology Manager's core logic for merging hints remains unchanged. However,
+its overall behavior will be updated through modifications to its `pod` and
+`container` scopes. These scopes will be enhanced to interact with hint
+providers (CPU and Memory Managers) that are now aware of `pod.spec.resources`,
+enabling more flexible and accurate NUMA alignment for pods utilizing this
+feature.
+
+#### Policies
+
+The behavior of existing Topology Manager policies and policy options will be
+supported for pod-level resources.
+
+-   **`none`**: This policy makes the Topology Manager idle. No changes are
+    needed.
+-   **`best-effort`**: This policy will admit pods regardless of whether the
+    NUMA alignment is preferred or not. It will be supported for pods with
+    pod-level resources.
+-   **`restricted`**: This policy will only admit pods if a preferred NUMA
+    alignment can be achieved. It will be supported for pods with pod-level
+    resources.
+-   **`single-numa-node`**: This policy will only admit pods if all of their
+    resources can be allocated from a single NUMA node. It will be supported for
+    pods with pod-level resources.
+
+#### Policy Options
+
+-   **`prefer-closest-numa-nodes`**: This option makes the Topology Manager
+    aware of NUMA distances when making alignment decisions. It will be
+    supported for pods with pod-level resources.
+-   **`max-allowable-numa-nodes`**: This policy allows the Topology manager to
+    work on nodes with more than eight NUMA nodes, and it will be supported by
+    pod level resources.
+
+#### Scopes
+
+##### Pod Scope
+
+When the `topology-manager-scope` is set to `pod`, the Topology Manager will
+invoke the `GetPodTopologyHints` function on its hint providers. This triggers
+the providers to generate a single set of hints based on the total resource
+budget defined in `pod.spec.resources`. This process ensures the entire pod,
+including containers with and without individual `Guaranteed` resource requests,
+can be aligned to a single NUMA node or a set of NUMA nodes.
+
+Pod-level resources | Container resources | CPU and Memory manager behavior
+:------------------ | :------------------ | :------------------------------
+Unset               | Set                 | Current behavior (alignment and exclusive allocation based on container resources)
+Set                 | Unset               | Alignment and exclusive allocation for the overall pod. All containers share the resulting resource pool.
+Set                 | Set                 | Alignment and exclusive allocation for the overall pod. The pod's resource pool is partitioned into exclusive slices for `Guaranteed` containers and a shared pool for the rest.
+
+When `pod.spec.resources` are defined, they are the sole determinant of the
+pod's QoS class. However, the resource managers will still perform a QoS-like
+check on each container to determine which are eligible for exclusive resource
```

Comment:
nit:  It might be good to explicitly state the criteria. 
```
The resource managers will still check container eligibility (request = limit, integer requests) for exclusive resource allocation.
```